### PR TITLE
Completed Emag homework with the required tests

### PR DESCRIPTION
**Pull request by sandrab28**
- test for filling in new account details
- test for adding item to cart
- test for adding items to favourites

### CODE REVIEW

File: src/test/java/steps/bws/ContactPageSteps.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public static void fillInSubjectInContactForm(String subject) {
+        fillWebElement(ContactPage.SUBJECT_FIELD, subject);
+    }
+
+    public static void fillInMessageInContactForm(String message) {
+        fillWebElement(ContactPage.MESSAGE_FIELD, message);
+    }
+
+    public static void displayCopyrightMessage() {
+        getTextFromWebElement(ContactPage.COPYRIGHT_MESSAGE);
```
Comment (BojteTamas) [inline src/test/java/steps/bws/ContactPageSteps.java:39]: here you only get the `CopyrightMessage` and it is not displayed

File: src/test/java/steps/emag/PracticeForSteps.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+
+public class PracticeForSteps extends WebDriverHelpers {
+
+    public static void deleteAllFavoriteProducts() {
+        List<WebElement> deleteButtons = getListOfElements(FavoritesPage.DELETE_FAVORITE_BUTTON);
+        for (WebElement button : deleteButtons) {
+            button.click();
+        }
+    }
```
Comment (BojteTamas) [inline src/test/java/steps/emag/PracticeForSteps.java:19]: Here you can make a general method, click all types of buttons or something like that

File: src/test/java/steps/emag/PracticeForSteps.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    public static void addAllFavoriteProductsToCart() {
+        List<WebElement> addToCartButtons = getListOfElements(FavoritesPage.ADD_TO_CART_BUTTON);
+        for (WebElement button : addToCartButtons) {
+            button.click();
+        }
+    }
```
Comment (BojteTamas) [inline src/test/java/steps/emag/PracticeForSteps.java:26]: And here we are, you can reuse the method from above

File: src/test/java/steps/emag/PracticeForSteps.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            clickWebElement(CommonPage.SEARCH_BUTTON);
+        }
+    }
+
+    public static void sendSmsLinkToPhoneNumber(int count) {
+        for (int i = 0; i < count; i++) {
+            fillWebElement(MainPage.PHONE_NUMBER_FIELD, RandomStringUtils.randomNumeric(5));
+            clickWebElement(MainPage.SEND_SMS_BUTTON);
+            clearWebElement(MainPage.PHONE_NUMBER_FIELD);
+        }
```
Comment (BojteTamas) [inline src/test/java/steps/emag/PracticeForSteps.java:40]: very nice

File: src/test/java/steps/facebook/LoginPageSteps.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    public static void pickYearOfBirth(String year) {
+        selectOptionFromDropdownByText(LoginPage.BIRTHDAY_YEAR, year);
+    }
+
+    public static void clickForgottenAccountLink() {
+        clickWebElement(LoginPage.FORGOTTEN_ACCOUNT_LINK);
+    }
+
+    public static void displayLanguagesList() {
```
Comment (BojteTamas) [inline src/test/java/steps/facebook/LoginPageSteps.java:49]: only get, not displayed

File: src/test/java/steps/bws/ContactPageSteps.java (hunk 1 above)
Comment (sandrab28) [reply src/test/java/steps/bws/ContactPageSteps.java]: Done, now using display method to display text from web element.

File: src/test/java/steps/facebook/LoginPageSteps.java (hunk 5 above)
Comment (sandrab28) [reply src/test/java/steps/facebook/LoginPageSteps.java:49]: Done, now using display method to display text from web element.

File: src/test/java/steps/emag/PracticeForSteps.java (hunk 2 above)
Comment (sandrab28) [reply src/test/java/steps/emag/PracticeForSteps.java]: Done, extracted general method

File: src/test/java/steps/emag/PracticeForSteps.java (hunk 3 above)
Comment (sandrab28) [reply src/test/java/steps/emag/PracticeForSteps.java]: Done

File: src/test/java/steps/emag/PracticeForSteps.java (hunk 4 above)
Comment (sandrab28) [reply src/test/java/steps/emag/PracticeForSteps.java:31]: thanks! :)

**Pull request merged**